Project/Src/Gui/TextAreaDragDropHandler.cs

```diff
@@ -93,24 +96,45 @@ protected void OnDragDrop(object sender, DragEventArgs e)
                     var offset = textArea.Caret.Offset;
                     if (textArea.IsReadOnly(offset))
                         return;
-                    if (e.Data.GetDataPresent(typeof(DefaultSelection)))
+
+                    try
                     {
-                        var sel = (ISelection)e.Data.GetData(typeof(DefaultSelection));
-                        if (sel.ContainsPosition(textArea.Caret.Position))
-                            return;
-                        if (GetDragDropEffect(e) == DragDropEffects.Move)
+                        if (e.Data.GetDataPresent(typeof(DefaultSelection)))
                         {
-                            if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                            var sel = (ISelection)e.Data.GetData(typeof(DefaultSelection));
+                            if (sel.ContainsPosition(textArea.Caret.Position))
                                 return;
-                            var len = sel.Length;
-                            textArea.Document.Remove(sel.Offset, len);
-                            if (sel.Offset < offset)
-                                offset -= len;
+                            if (GetDragDropEffect(e) == DragDropEffects.Move)
+                            {
+                                if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                                    return;
+                                var len = sel.Length;
+                                textArea.Document.Remove(sel.Offset, len);
+                                if (sel.Offset < offset)
+                                    offset -= len;
+                            }
                         }
                     }
-
+                    catch (System.InvalidCastException)
+                    {
+                    /*
+                        If GetDataPresent(typeof(DefaultSelection)) threw this
+                        exception, then it's an interprocess DefaultSelection
+                        COM object that is not serializable! In general,
+                        GetDataPresent(typeof(...)) throws InvalidCastException
+                        for drags and drops from other GitExt processes (maybe
+                        we need to make the data objects [Serializable]?). We
+                        can get around this exception by doing
+                        GetDataPresent(String s) [using the string of the type
+                        name seems to work fine!] Since it is interprocess
+                        data, just get the string data from it - special
+                        handling logic in try {} is only valid for selections
+                        within the current process's text editor!
+                    */
```

Comment:
Insufficient indentation

Reply:
@RussKie @gerhardol 
Here, I am adhering to the 80 character horizontal limit.  One more tab and there will be very little text per line, and the comment will span a few more lines.  Is it fine to add tabs and go with a max of 84 characters horizontally, rather than 80 characters?

Reply:
We need to mindful of line lengths, but I wouldn't religiously adhere to 80char rule. We don't work on 80x25 terminals, it is quite common for devs to work on ultra-wide or 4K monitors.

Reply:
I used to set margins at 150/180 chars, I think that fit perfectly on 1920x screens.

@mstv do you work on a small resolution laptop? How many chars per line can you comfortably see at 100%?

Reply:
I used to set margins at 150/180 chars, I think that fit perfectly on 1920x screens.

@mstv do you work on a small resolution laptop? How many chars per line can you comfortably see at 100%?

Reply:
> Do you work on a small resolution laptop? How many chars per line can you comfortably see at 100%?

No I don't and won't 😄. My former comment was about the Settings dialog, which does not resize well enough. It does not need more than 740 lines. So we should not make it taller.

Reply:
I requested a change for this when the comment length was like 160 chars (+ indent), which is very hard to read in GitHub, we should be able to review there at least. 
Up to 100 chars is not a big problem, in this case the comments as a whole will be readable when side scrolling some.

As many lines are over 100 chars in the existing code, I have to resort to two editors side by side, but I would have preferred max 100 chars and three editor panels.

Reply:
You really need to install [Wide Github](https://chrome.google.com/webstore/detail/wide-github/kaalofacklcidaampbokdplbklpeldpj?hl=en) 😎

Reply:
This is how it looks for me on a MBP
![image](https://user-images.githubusercontent.com/4403806/68811581-16db0d80-06c5-11ea-8e1c-39a990c259fa.png)

Reply:
Does not work on FireFox
There migfht be something similar for FF
But such measure should not be needed just to view

Reply:
https://addons.mozilla.org/en-US/firefox/addon/widegithub/

Reply:
This is the remaining open comment. I suggest adding a tab and we are done with it.
Squash as well.